x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/AbstractConfigurationFunctionTestCase.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.expression.function.scalar;
+
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.xpack.esql.EsqlTestUtils;
+import org.elasticsearch.xpack.esql.expression.function.AbstractFunctionTestCase;
+import org.elasticsearch.xpack.esql.plugin.EsqlPlugin;
+import org.elasticsearch.xpack.esql.plugin.QueryPragmas;
+import org.elasticsearch.xpack.esql.session.EsqlConfiguration;
+import org.elasticsearch.xpack.ql.expression.Expression;
+import org.elasticsearch.xpack.ql.tree.Source;
+import org.elasticsearch.xpack.ql.util.StringUtils;
+
+import java.util.List;
+
+import static org.elasticsearch.xpack.esql.SerializationTestUtils.assertSerialization;
+
+public abstract class AbstractConfigurationFunctionTestCase extends AbstractFunctionTestCase {
```

Comment:
@not-napoleon maybe we should (separately to this PR) add test cases for functions that rely on the `EsqlConfig`. E.g. `NOW` actually derives its value from the config, which maybe could use a nice randomized test.

Reply:
That's a good point.  Let's get a ticket open for that, so we don't forget.  for `NOW` at least, it might make sense for it to not use `AbstractFunctionTestCase` at all, since it doesn't really have any parameters we need to randomize, null, type check, etc.

Reply:
Here we go: https://github.com/elastic/elasticsearch/issues/107997